lib/cohere-embed-v3-services-stack.ts

```diff
@@ -19,11 +19,9 @@ export class CohereEmbedV3ServicesStack extends cdk.Stack {
   constructor(scope: Construct, id: string, props: CohereEmbedV3ServicesStackProps) {
     super(scope, id, props);
 
-    const vpcStack = new CohereEmbedVpcStack(this, `${props.resourcePrefix}-VpcStack`, {
-      ...props,
-      description: 'VPC for Cohere Embed V3.',
+    const vpc = ec2.Vpc.fromLookup(this, `${props.resourcePrefix}-VPC`, {
+      vpcId: props.vpcId,
     });
```

Comment:
**Suggestion:** Consider adding a check to ensure that the VPC has the necessary subnets and configuration required for your application. This can prevent deployment issues caused by incompatible VPC configurations.
> 🏜️  **High Priority** 
 > [Category: **`best practice`**, Importance: **`8`**]
> **Reason:** This suggestion adds a crucial check for VPC configuration, which can prevent deployment issues and ensure the application has the required network resources.

```suggestion
    const vpc = ec2.Vpc.fromLookup(this, `${props.resourcePrefix}-VPC`, {
      vpcId: props.vpcId,
    });

    if (vpc.publicSubnets.length === 0 || vpc.privateSubnets.length === 0) {
      throw new Error(`The VPC ${props.vpcId} does not have both public and private subnets. Please ensure the VPC is properly configured.`);
    }
```